services/orion-decision/src/orion_decision/ci_matrix.py

```diff
@@ -227,16 +236,23 @@ def from_json(cls, data: str) -> "MatrixJob":
         result.stage = obj["stage"]
         result.require_previous_stage_pass = obj["require_previous_stage_pass"]
         result.secrets.extend(CISecret.from_json(secret) for secret in obj["secrets"])
+        result.artifacts.extend(
+            CIArtifact.from_json(artifact) for artifact in obj.get("artifacts", [])
+        )
         result.check()
         return result
 
     def __eq__(self, other) -> bool:
         for attr in self.__slots__:
             if attr == "secrets":
                 continue
+            if attr == "artifacts":
+                continue
             if getattr(self, attr) != getattr(other, attr):
                 return False
-        return all(secret in other.secrets for secret in self.secrets)
+        return all(secret in other.secrets for secret in self.secrets) and all(
```

Comment:
This does not check if other contains items not in self, not sure if that matters.

Reply:
This is true and applies to secrets also. I'll add another commit to fix.